src/components/SwitchButton/index.tsx

```diff
@@ -1,204 +1,141 @@
+    top: 4px;
+    left: 4px;
+    width: calc(
+      50% - ${({ theme }) => theme.space['1']} -
+        ${({ theme }) => theme.space['0.5']}
+    );
+    height: calc(100% - ${({ theme }) => theme.space['1']});
+    border-radius: ${({ theme }) => theme.radii.default};
+    background-color: ${({ theme }) => theme.colors.primary.backgroundStrong};
+    transition: transform 300ms, width 300ms, left 300ms;
   }
 
-  &[aria-disabled='true'] {
-    cursor: not-allowed;
-    color: ${({ theme }) => theme.colors.neutral.textDisabled};
-    background-color: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
-    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
-    pointer-events: none;
+  &[data-position='right']:before {
+    left: calc(100% - 4px);
+    width: 50%;
+    transform: translateX(-100%);
+  }
 
-    &[data-checked='true'] {
-      color: ${({ theme }) => theme.colors.neutral.textDisabled};
-      border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
-    }
+  &[data-position='right']:active:before {
+    width: calc(50% + ${({ theme }) => theme.space['0.5']});
   }
 
-  &:not([data-checked='true']):not([aria-disabled='true']) {
-    &:hover,
-    &:focus {
-      border-color: ${({ theme }) => theme.colors.primary.border};
-    }
+  &[data-position='left']:active:before {
+    width: calc(50% - ${({ theme }) => theme.space['0.5']} - 1px);
+  }
+`
 
-    &:hover {
-      box-shadow: ${({ theme }) => theme.shadows.hoverPrimary};
-    }
+const StyledSelectableCard = styled(SelectableCard)<{ checked: boolean }>`
+  border: none;
+  height: 40px;
+  padding: ${({ theme }) => theme.space['1']} ${({ theme }) => theme.space['2']};
+  font-weight: ${({ theme }) => theme.typography.bodyStrong.weight};
+  justify-content: center;
+  align-items: center;
 
-    &:focus {
-      box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
-    }
+  &:hover,
+  &:focus-within,
+  &:active {
+    box-shadow: none;
+    border: none;
   }
 
-  ${({ theme }) => Object.values(variants).map(variantFn => variantFn(theme))}
-`
-
-const StyledRadio = styled(Radio)`
-  position: absolute;
-  opacity: 0.01;
+  ${({ checked, theme }) =>
+    checked
+      ? `
+  color: ${theme.colors.primary.textStrong};
+  `
+      : `
+        &:hover {
+          color: ${theme.colors.primary.textWeak};
+        }
+      `}
 `
 
-type SwitchButtonProps = Omit<StyledSwitchProps, 'onChange' | 'children'> & {
-  children:
-    | ReactNode
-    | (({
-        checked,
-        disabled,
-      }: {
-        checked: boolean
-        disabled: boolean
-      }) => ReactNode)
+type SwitchButtonProps = {
   name: string
   onBlur?: FocusEventHandler
   onChange: ChangeEventHandler
   onFocus?: FocusEventHandler
   tooltip?: string
-  value: string
-  checked?: boolean
-  disabled?: boolean
-  variant?: Variants
+  value?: string | number
+  leftValue: string | number
+  rightValue: string | number
+  leftText: string
+  rightText: string
```

Comment:
I think it would be a better DX to have label-value pairs. But it's up to you!

```suggestion
  leftButton: {
    label: string,
    value: string
  },
  rightButton: {
    label: string,
    value: string
  }
```

Reply:
@baptisteArno I like your idea